modules/tinymce/src/core/main/ts/fmt/ApplyFormat.ts

```diff
@@ -292,18 +302,23 @@ const applyFormat = (ed: Editor, name: string, vars?: FormatVars, node?: Node |
     });
   };
 
-  if (dom.getContentEditable(selection.getNode()) === 'false') {
-    node = selection.getNode();
+  const selectedNode = selection.getNode();
+  if (dom.getContentEditable(selectedNode) === 'false') {
+    node = selectedNode;
+    let formatApplied = false;
     for (let i = 0, l = formatList.length; i < l; i++) {
       const formatItem = formatList[i];
       if (formatItem.ceFalseOverride && FormatUtils.isSelectorFormat(formatItem) && dom.is(node, formatItem.selector)) {
         setElementFormat(node as Element, formatItem);
+        formatApplied = true;
         break;
       }
     }
 
-    Events.fireFormatApply(ed, name, node, vars);
-    return;
+    if (formatApplied || !FormatUtils.isWrappableNoneditable(ed, node)) {
+      Events.fireFormatApply(ed, name, selectedNode, vars);
+      return;
+    }
```

Comment:
This looks to be a breaking change... as this will make the event change from any `editor.formatter.apply()` call to only firing when a format was applied.

Are we sure we want to do this? If so then it definitely needs a changelog entry.

Reply:
The event should still be fired by the `Events.fireFormatApply` call near the end of the file even if a format wasn't applied. I could be misunderstanding something though.

Reply:
It won't look at your logic... it'll only fire if a format applied now or it's not a non-wrappable noneditable node. So if it's a wrappable non-editable element and a format doesn't match then it no longer fires. Basically we don't want these if conditions.

Reply:
Ahh wait, I see what you mean now. There is a second/separate call at the bottom of this. Hmm... I'll need to look more but this does worry me.

Reply:
I have refactored and removed a fair bit of code in both ApplyFormat.ts and RemoveFormat.ts to remove the need to have the specific logic for when a non-editable element is selected. There is a risk by doing this but all of the tests pass which is a pretty good sign given there are a lot of formatted tests.